File: runme/mcgen/local/datan/datsrc/sqftst.c
#include "datsrc.h"
 
double LIBFUNC sqftst_(double const P_T *p, integer const P_T *n1,
                      integer const P_T *n2)
{
    /* Table of constant values */

    double c_d1em6 = 1e-6;

    /* System generated locals */
    double ret_val;

    /* Local variables */
    double xzero, x0, x1;

/* boundary of range */
    if (*p >= 1.) {
        ret_val = 1e10;
    }
    if (*p <= 0.) {
        ret_val = 0.;
    }
/* normal range */
    x0 = 0.;
    x1 = *p;
    if (*p < 1. && *p > 0.) {
        auxzbr_(&x0, &x1, szftst_, p, n1, n2);
        auxzfn_(&x0, &x1, &xzero, szftst_, p, n1, n2, &c_d1em6);
        ret_val = xzero;
    }
    return ret_val;
} /* sqftst_ */

/* ----------------------------------------------------------------- */
double LIBFUNC szftst_(double const P_T *x, double const P_T *p,
                      integer const P_T *n1, integer const P_T *n2)
{
    /* System generated locals */
    double ret_val;

    /* Local variables */

/* returns P minus cumulative F-distribution of (X,N1,N2) */
    ret_val = *p - scftst_(x, n1, n2);
    return ret_val;
} /* szftst_ */

File: runme/mcgen/local/datan/datsrc/sqftst.f
      DOUBLE PRECISION FUNCTION SQFTST(P,N1,N2)
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      PARAMETER (BIG=1D10, EPSILN=1D-6, ONE=1.D0, ZERO=0.D0)
      EXTERNAL SZFTST
C boundary of range
      IF(P.GE.ONE) SQFTST=BIG
      IF(P.LE.ZERO) SQFTST=ZERO
C normal range
      X0=ZERO
      X1=P
      IF(P.LT.ONE .AND. P.GT.ZERO) THEN
        CALL AUXZBR(X0,X1,SZFTST,P,N1,N2)
        CALL AUXZFN(X0,X1,XZERO,SZFTST,P,N1,N2,EPSILN)
        SQFTST=XZERO
      END IF
      END
C-----------------------------------------------------------------
      DOUBLE PRECISION FUNCTION SZFTST(X,P,N1,N2)
C returns P minus cumulative F-distribution of (X,N1,N2)
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      SZFTST=P-SCFTST(X,N1,N2)
      END
